zhaquirks/tuya/ts0601_trv_sas.py

```diff
@@ -192,9 +217,54 @@ class Thermostat_TYST11_c88teujp(TuyaThermostat):
     }
 
 
+class Thermostat_TZE200_c88teujp_TemperatureOffset(LocalDataCluster, AnalogOutput):
+    """AnalogOutput cluster for setting temperature offset."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.temperature_calibration_bus.add_listener(self)
+        self._update_attribute(
+            self.attributes_by_name["description"].id, "Temperature Offset"
+        )
+        self._update_attribute(self.attributes_by_name["max_present_value"].id, 6)
+        self._update_attribute(self.attributes_by_name["min_present_value"].id, -6)
+        self._update_attribute(self.attributes_by_name["resolution"].id, 1)
+        self._update_attribute(self.attributes_by_name["application_type"].id, 0x0009)
+        self._update_attribute(self.attributes_by_name["engineering_units"].id, 62)
+
+    def set_value(self, value):
+        """Set new temperature offset value."""
+        self._update_attribute(self.attributes_by_name["present_value"].id, value)
+
+    def get_value(self):
+        """Get current temperature offset value."""
+        return self._attr_cache.get(self.attributes_by_name["present_value"].id)
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Modify value before passing it to the set_data tuya command."""
+        for attrid, value in attributes.items():
+            if isinstance(attrid, str):
+                attrid = self.attributes_by_name[attrid].id
+            if attrid not in self.attributes:
+                self.error("%d is not a valid attribute id", attrid)
+                continue
+            self._update_attribute(attrid, value)
+
+            await self.endpoint.tuya_manufacturer.write_attributes(
+                {SASWELL_TEMPERATURE_CALIBRATION_ATTR: value}, manufacturer=None
+            )
```

Comment:
That will update `SASWELL_TEMPERATURE_CALIBRATION_ATTR` for any cluster attribute being updated.
Validate `attriid == 0x0055` (`present_value`)

Reply:
If I understand you right, this will overwrite ``attriid = 0x0055`` with  ``attriid = 0x38`` from the quirk? I wouldn't know how to address this. Delete the statement?

Reply:
The call is fine, the problem would be when is it called.

The `TuyaTemperatureOffset` would have multiple attributes (all the ones from [AnalogInput](https://github.com/zigpy/zigpy/blob/8af92dee9182ce54966621da1d0125e67b8c72f5/zigpy/zcl/clusters/general.py#L1167)).

But you only wants to call this part if the updated attribute is the `present_value`:
```python
            await self.endpoint.tuya_manufacturer.write_attributes(
                {SASWELL_TEMPERATURE_CALIBRATION_ATTR: value}, manufacturer=None
            )
```
and not for any other attribute.

Usually you would wrap the code with a validation like:
```python
        if attrid == 0x0055:  # `present_value`
                await self.endpoint.tuya_manufacturer.write_attributes(
                    {SASWELL_TEMPERATURE_CALIBRATION_ATTR: value}, manufacturer=None
                )
```

Reply:
I added your suggested validation.